nn_surrogate_benchmark/surrogate_sobol.py

```diff
@@ -0,0 +1,209 @@
+import torch
+import torch.nn as nn
+import pytorch_lightning as pl
+from torch.optim import AdamW, Adam
+from typing import Literal
+import pandas as pd
+import numpy as np
+from torch.utils.data import TensorDataset, DataLoader
+from sklearn.preprocessing import (
+	StandardScaler,
+	MinMaxScaler,
+	RobustScaler,
+	MaxAbsScaler,
+)
+
+class Sobolev(pl.LightningModule):
+	def __init__(
+		self,
+		input_dim: int,
+		hidden_dims: list[int] = [512],
+		activation: Literal["tanh", "relu", "gelu"] = "tanh",
+		lam1: float = 1.0,  # Weighting factor for gradient loss
+		lr = 3e-4,
+	) -> None:
+		super().__init__()
+		self.save_hyperparameters()
+
+		layers = []
+		for hidden_dim in hidden_dims:
+			layers.append(nn.Linear(input_dim, hidden_dim))
+			if activation == "tanh":
+				layers.append(nn.Tanh())
+			elif activation == "relu":
+				layers.append(nn.ReLU(inplace=False))  # Ensure inplace=False
```

Comment:
why `inplace=False`? we don't have this for MLP

Reply:
Yeah, it's unneeded - inplace=False is the default. Removed it